experimental/InvariantTheory/invariant_rings.jl

```diff
@@ -716,21 +716,42 @@ AbstractAlgebra.Generic.MatSpaceElem{nf_elem}[[0 0 1; 1 0 0; 0 1 0], [1 0 0; 0 a
 
 julia> fundamental_invariants(IR)
 4-element Vector{MPolyElem_dec{nf_elem, AbstractAlgebra.Generic.MPoly{nf_elem}}}:
- x[1]*x[2]*x[3]
  x[1]^3 + x[2]^3 + x[3]^3
- x[1]^3*x[2]^3 + x[1]^3*x[3]^3 + x[2]^3*x[3]^3
+ x[1]*x[2]*x[3]
+ x[1]^6 + x[2]^6 + x[3]^6
  x[1]^6*x[3]^3 + x[1]^3*x[2]^6 + x[2]^3*x[3]^6
 ```
 """
-function fundamental_invariants(IR::InvRing)
+function fundamental_invariants(IR::InvRing, algo::Symbol = :king)
   if !isdefined(IR, :fundamental)
-     V = primary_invariants(IR)
-     append!(V, irreducible_secondary_invariants(IR))
-     IR.fundamental = minimal_subalgebra_generators(V)
+    if algo == :king
+      IR.fundamental = fundamental_invariants_via_king(IR)
+    elseif algo == :minimal_subalgebra
+      IR.fundamental = fundamental_invariants_via_minimal_subalgebra(IR)
+    else
+      error("Unsupported argument :$(algo) for algo")
+    end
   end
   return IR.fundamental
 end
 
+function fundamental_invariants_via_minimal_subalgebra(IR::InvRing)
+  V = primary_invariants(IR)
+  append!(V, irreducible_secondary_invariants(IR))
+  return minimal_subalgebra_generators(V)
+end
+
+function fundamental_invariants_via_king(IR::InvRing)
+  rey = reynolds_via_singular(IR)
+  F = Singular.LibFinvar.invariant_algebra_reynolds(rey)
```

Comment:
If you know the return typeof `LibFinvar.invariant_algebra_reynolds`, the compiler would probably thank you for a type assertion here, as it has absolutely no idea what is going to be returned.

Reply:
Good point.
I fixed it a little bit, but getting the correct type is difficult...

Reply:
This is fine now.